yt/data_objects/selection_data_containers.py

```diff
@@ -587,6 +585,13 @@ class YTDisk(YTSelectionContainer3D):
     _con_args = ('center', '_norm_vec', 'radius', 'height')
     def __init__(self, center, normal, radius, height, fields=None,
                  ds=None, field_parameters=None, data_source=None):
+
+        validate_3d_array(center, normal)
+        validate_float(radius, height)
+        validate_iterable(fields)
+        validate_object((field_parameters, dict),
+                        (ds, Dataset), (data_source, YTSelectionContainer))
```

Comment:
This API seems much better, although the last call is pretty hard to read. Did we lose the validation that the arrays are the same length?

Reply:
Any suggestion to make the `validate_object` call better? Let me know if array length validation ( len=3) has to be enforced too in `validate_3d_array`.